crates/ide_completion/src/completions/record.rs

```diff
@@ -45,9 +47,102 @@ pub(crate) fn complete_record(acc: &mut Completions, ctx: &CompletionContext) ->
     Some(())
 }
 
+pub(crate) fn complete_record_literal(
+    acc: &mut Completions,
+    ctx: &CompletionContext,
+) -> Option<()> {
+    if let hir::Adt::Struct(strukt) = ctx.expected_type.as_ref()?.as_adt()? {
+        acc.add_struct_literal(ctx, strukt, None);
+    }
+
+    Some(())
+}
+
 #[cfg(test)]
 mod tests {
-    use crate::tests::check_edit;
+    use crate::{
+        item::CompletionKind,
+        tests::{check_edit, do_completion},
+    };
+
+    #[test]
+    fn literal_struct_completion_edit() {
+        check_edit(
+            "FooDesc {…}",
+            r#"
+struct FooDesc { pub bar: bool }
+
+fn create_foo(foo_desc: &FooDesc) -> () { () }
+
+fn baz() {
+    let foo = create_foo(&$0);
+}
+            "#,
+            r#"
+struct FooDesc { pub bar: bool }
+
+fn create_foo(foo_desc: &FooDesc) -> () { () }
+
+fn baz() {
+    let foo = create_foo(&FooDesc { bar: ${1:()} }$0);
+}
+            "#,
+        )
+    }
+
+    #[test]
+    // If all fields are public a completion is added, otherwise not.
+    fn literal_struct_complexion_private_fields() {
+        // `FooDesc.bar` is private, the completion should not be triggered.
+        let code = r#"
+mod _69latrick {
+    pub struct FooDesc { pub six: bool, pub neuf: Vec<String>, bar: bool }
+    pub fn create_foo(foo_desc: &FooDesc) -> () { () }
+}
+
+fn baz() {
+    use _69latrick::*;
+
+    let foo = create_foo(&$0);
+}
+            "#;
+
+        let nbr_of_completion_matching = do_completion(code, CompletionKind::Snippet)
+            .into_iter()
+            .filter(|item| item.lookup() == "FooDesc {…}")
+            .count();
+
+        stdx::never!(nbr_of_completion_matching != 0);
```

Comment:
You can move this part into a test in `ide_completions/src/expressions.rs` instead, there we can then see whether the completion appears in the `expect!` output or not.